dataprofiler/profilers/histogram_utils.py

```diff
@@ -14,6 +14,293 @@
+    """
+    iqr = np.subtract(profile._get_percentile([75]), profile._get_percentile([25]))
+    try:
+        dataset_size = profile.match_count
+    except AttributeError:
+        dataset_size = sum(profile._stored_histogram["histogram"]["bin_counts"])
+    return 2.0 * iqr * dataset_size ** (-1.0 / 3.0)
+
+
+def _calc_auto_bin_width_from_profile(profile):
+    """
+    Histogram bin estimator that uses Freedman-Diaconis and Sturges estimators.
+
+    Chooses the minimum width of the estimators if the FD bin width is non-zero.
+    If the bin width from the FD estimator is 0, the Sturges estimator is used.
+
+    The FD estimator is usually the most robust method, but its width
+    estimate tends to be too large for small `x` and bad for data with limited
+    variance. The Sturges estimator is quite good for small (<1000) datasets
+    and is the default in the R language. This method gives good off-the-shelf
+    behaviour.
+
+    Function follows the numpy implementation within:
+    https://github.com/numpy/numpy/blob/main/numpy/lib/histograms.py
+
+    Parameters
+    ----------
+    profile : NumericStatsMixin
+        Input data's data profile that is to be histogrammed
+
+    Returns
+    -------
+    An estimate of the optimal bin width for the given data.
+
+    See Also
+    --------
+    _hist_bin_fd, _hist_bin_sturges
+    """
+    fd_bw = _calc_fd_bin_width_from_profile(profile)
+    sturges_bw = _calc_sturges_bin_width_from_profile(profile)
+    if fd_bw:
+        return min(fd_bw, sturges_bw)
+    else:
+        # limited variance, so we return a len dependent bw estimator
+        return sturges_bw
+
+
+def _calc_scott_bin_width_from_profile(profile):
+    """
+    Scott histogram bin estimator reworked to use profiles.
+
+    The binwidth is proportional to the standard deviation of the data
+    and inversely proportional to the cube root of data size
+    (asymptotically optimal).
+
+    Function follows the numpy implementation within:
+    https://github.com/numpy/numpy/blob/main/numpy/lib/histograms.py
+
+    Parameters
+    ----------
+    profile : NumericStatsMixin
+        Input data's data profile that is to be histogrammed
+
+    Returns
+    -------
+    h : An estimate of the optimal bin width for the given data.
+    """
+    try:
+        dataset_size = profile.match_count
+    except AttributeError:
+        dataset_size = sum(profile._stored_histogram["histogram"]["bin_counts"])
+    std = profile.stddev
+    return (24.0 * np.pi**0.5 / dataset_size) ** (1.0 / 3.0) * std
+
+
+_hist_bin_width_selectors_for_profile = {
+    "auto": _calc_auto_bin_width_from_profile,
+    "doane": _calc_doane_bin_width_from_profile,
+    "fd": _calc_fd_bin_width_from_profile,
+    "rice": _calc_rice_bin_width_from_profile,
+    "scott": _calc_scott_bin_width_from_profile,
+    "sqrt": _calc_sqrt_bin_width_from_profile,
+    "sturges": _calc_sturges_bin_width_from_profile,
+}
+
+
```

Comment:
at this point in the code, have already calc'ed the values necessary to generate this? as if we did it from the profile? This would be __calculations order dependent.

Reply:
could be a refactor to remove.

Reply:
nvm, maybe we cannot bc initial of hist not yet created for quantiles?